indico/migrations/versions/20250224_1832_4615aff776e0_add_user_logs.py

```diff
@@ -0,0 +1,66 @@
+"""Add user logs
+
+Revision ID: 4615aff776e0
+Revises: 9251bc3e2106
+Create Date: 2025-02-24 18:32:20.784460
+"""
+
+from enum import Enum
+
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy.dialects import postgresql
+
+from indico.core.db.sqlalchemy import PyIntEnum, UTCDateTime
+
+
+# revision identifiers, used by Alembic.
+revision = '4615aff776e0'
+down_revision = '9251bc3e2106'
+branch_labels = None
+depends_on = None
+
+
+class _UserLogRealm(int, Enum):
+    user = 1
+
+
+class _LogKind(int, Enum):
+    other = 1
+    positive = 2
+    change = 3
+    negative = 4
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table('logs',
+        sa.Column('id', sa.Integer(), nullable=False),
+        sa.Column('logged_dt', UTCDateTime, nullable=False),
+        sa.Column('kind', PyIntEnum(_LogKind), nullable=False),
+        sa.Column('module', sa.String(), nullable=False),
+        sa.Column('type', sa.String(), nullable=False),
+        sa.Column('summary', sa.String(), nullable=False),
+        sa.Column('data', postgresql.JSON(astext_type=sa.Text()), nullable=False),
+        sa.Column('meta', postgresql.JSONB(astext_type=sa.Text()), nullable=False),
+        sa.Column('target_user_id', sa.Integer(), nullable=False),
+        sa.Column('realm', PyIntEnum(_UserLogRealm), nullable=False),
+        sa.Column('user_id', sa.Integer(), nullable=True),
+        sa.ForeignKeyConstraint(['target_user_id'], ['users.users.id'], name=op.f('fk_logs_target_user_id_users')),
+        sa.ForeignKeyConstraint(['user_id'], ['users.users.id'], name=op.f('fk_logs_user_id_users')),
+        sa.PrimaryKeyConstraint('id', name=op.f('pk_logs')),
+        schema='users'
+    )
+    op.create_index(op.f('ix_logs_meta'), 'logs', ['meta'], unique=False, schema='users', postgresql_using='gin')
+    op.create_index(op.f('ix_logs_target_user_id'), 'logs', ['target_user_id'], unique=False, schema='users')
+    op.create_index(op.f('ix_logs_user_id'), 'logs', ['user_id'], unique=False, schema='users')
```

Comment:
Check e.g. `20210802_1514_4b097412a8d9_add_category_logs.py` to see how this can be simplified (`index=True` on the columns)